src/dmd/parse.d

```diff
@@ -5533,7 +5533,18 @@ final class Parser(AST) : Lexer
         Lexp:
             {
                 AST.Expression exp = parseExpression();
-                check(TOK.semicolon, "statement");
+                /* https://issues.dlang.org/show_bug.cgi?id=15103
+                 * Improve declaration / initialization syntax error message
+                 * Error: found 'foo' when expecting ';' following statement
+                 * becomes Error: found `(` when expecting `;` or `=`, did you mean `Foo foo = 42`?
+                 */
+                if (token.value == TOK.identifier)
```

Comment:
```suggestion
                if (token.value == TOK.identifier && exp.op == TOK.identifier)
```

Reply:
Then revert the message change in fail196.d

Reply:
Thank you for your feedback! I modified.